### PR TITLE
tests: add unit tests for utils/helpers

### PR DESCRIPTION
**Pull request by jacquerie**
* Adds unit tests for methods in `inspirehep/utils/helpers.py`.

~~Depends on #1752~~
Spin off of #1748

### CODE REVIEW

**Comment by jacquerie** [comment]
Ready for review.

File: tests/integration/test_pidstore.py [hunk 1]
```diff
@@ -24,11 +24,13 @@
 
 import httpretty
 import mock
+import pytest
 
 from inspirehep.modules.pidstore.providers import InspireRecordIdProvider
 
 
-def test_getting_next_recid_from_legacy(httpretty_mock, app):
+@pytest.mark.xfail
```
Comment (jacquerie) [inline tests/integration/test_pidstore.py]: Whoops. Wrong `mark`.

File: tests/unit/utils/test_utils_helpers.py [hunk 2]
```diff
@@ -20,7 +20,140 @@
...
+    filename = next(tempfile._get_candidate_names())
+
+    expected = filename
+    result = download_file(
+        'http://example.com/500', output_file=filename)
+
+    assert expected == result
+
+    with pytest.raises(IOError):
+        open(filename, 'rb')
```
Comment (jacquerie) [inline tests/unit/utils/test_utils_helpers.py]: @kaplun This documents this surprising fact: if the request fails, no file is created, but no exception is thrown. I don't think this is good behavior.

File: tests/unit/utils/test_utils_helpers.py [hunk 3]
```diff
@@ -20,7 +20,140 @@
...
+
+
+@pytest.mark.httpretty
+def test_download_file_raises_if_called_without_output_file():
+    httpretty.register_uri(
+        httpretty.GET, 'http://example.com/foo-bar-baz', body=foo_bar_baz)
+
+    with pytest.raises(IOError):
+        download_file('http://example.com/foo-bar-baz')
```
Comment (jacquerie) [inline tests/unit/utils/test_utils_helpers.py]: This instead documents the weird `None` default argument, which I don't think it's a good idea either.

**Comment by kaplun** [comment]
@jacquerie since the immediate objective is to deliver all this code on production, and you seem to have a found quite a source of bugs (or potential source of bugs), please send an appropriate fix as well.

In particular, raising an exception that should rather being captured upstream (e.g. in the workflow, cc @david-caro), and passing around the open socket. Or rather pre-downloading the file locally and only afterwards move-it around.

**Comment by jacquerie** [comment]
These tests are blocked by the fact that I don't know what is the correct behaviour: attaching a string containing the file contents to the `files`, or an **open** file descriptor. We agreed that the former is better, but in some places of the code the latter is expected.

**Comment by kaplun** [comment]
I thought we agreed that the safest thing is actually to download it locally (better on EOS) and ship around the full path to the file?

**Comment by jacquerie** [comment]
> I thought we agreed that the safest thing is actually to download it locally (better on EOS) and ship around the full path to the file?

This is not at all what the users of `.files` expect, so they should be amended accordingly. To be clear, what you're proposing is that `.files` becomes a map of filename -> path on EOS, right?

**Comment by jacquerie** [comment]
> To be clear, what you're proposing is that `.files` becomes a map of filename -> path on EOS, right?

This requires a modification to https://github.com/inveniosoftware-contrib/invenio-workflows-files/blob/22570da9aeb6b8029e2f88a3a5379d28a4f6b9ee/invenio_workflows_files/api.py.

**Comment by david-caro** [comment]
@jacquerie is there an issue I can assign myself to?

**Comment by jacquerie** [comment]
> @jacquerie is there an issue I can assign myself to?

I was just writing it!

**Comment by jacquerie** [comment]
Me and @david-caro  just figured out that the treatment of files is way more busted than what these tests say. Writing the issue will have to wait until we understand a little more what's going on.

**Comment by kaplun** [comment]
For this milestone there minimal viable product is:
* The FFT in the MARCXML that is sent to legacy either contain the URL of the arXiv paper
* Alternatively the FFT refers to a local file that is accessible via EOS by legacy.

Pick up what is less broken

**Comment by kaplun** [comment]
> Writing the issue will have to wait until we understand a little more what's going on.

At least write a stub issue with the right milestone, so that everybody else is aware that someone is working on something that is a blocker.

**Pull request merged**